### ISSUE TITLE
sno show/diff doesn't work for initial import commits

### ISSUE THREAD

**Issue by craigds** [koordinates/sno#44]
There doesn't seem to be a way to view the content of an import commit.

Behold a fledgling sno repository:

```
$ sno log
commit 736a79dcb844a55e88813635b3535dd227635a87 (HEAD -> master)
Author: Craig de Stigter <craig@destigter.nz>
Date:   Tue Apr 7 13:55:50 2020 +1200

    Import from manawatu-whanganui-palmerston-north-lidar-index-tiles-2018.gpkg to /manawatu_whanganui_palmerston_north_lidar_index_tiles_2018/
```

None of the following commands produce much to show the commit's contents:

```
cdestigter@behindyou  $ sno diff 736a79dcb844a55e88813635b3535dd227635a87
```

```
cdestigter@behindyou  $ sno diff HEAD^..HEAD
Usage: sno diff [OPTIONS] [ARGS]...

Error: Invalid value for commit: Invalid commit spec
```

```
cdestigter@behindyou  $ sno show
commit 736a79dcb844a55e88813635b3535dd227635a87 (HEAD -> master)
Author: Craig de Stigter <craig@destigter.nz>
Date:   Tue Apr 7 13:55:50 2020 +1200

    Import from manawatu-whanganui-palmerston-north-lidar-index-tiles-2018.gpkg to /manawatu_whanganui_palmerston_north_lidar_index_tiles_2018/
```

**Comment by craigds** [comment]
the linked PR doesn't actually quite fix this yet - we still need to resolve `X^` to the empty-tree ref everywhere, when X refers to the first commit.

**Comment by olsen232** [comment]
X^ still doesn't refer to the empty tree everywhere - git doesn't actually do this either. Not sure if it's a good behaviour to implement or not.

But, git show <initial_commit> now shows the initial commit including a diff <empty_tree>..<initial_commit>. So that part works

**Comment by craigds** [comment]
👍 fixed via 222f1ceaeccc36aef19f8aca3edf23d6cb738d19
